test/plugin/base_test.rb

```diff
@@ -1025,17 +1025,26 @@ def test_http_request_partial_from_record
     end
   end
 
-  def test_http_request_without_referer_from_record
+  def test_http_request_from_record_with_referer_nil_or_absent
     setup_gce_metadata_stubs
-    setup_logging_stubs do
-      d = create_driver
-      d.emit('httpRequest' => http_request_message_without_referer)
-      d.run
-    end
-    verify_log_entries(1, COMPUTE_PARAMS, 'httpRequest') do |entry|
-      assert_equal http_request_message_without_referer, entry['httpRequest'],
-                   entry
-      assert_nil get_fields(entry['structPayload'])['httpRequest'], entry
+    {
+      http_request_message_with_referer_nil => \
+        http_request_message_expected_struct_payload,
+      http_request_message_with_referer_absent => 'null'
+    }.each do |message, expected|
+      @logs_sent = []
+      setup_logging_stubs do
+        d = create_driver
+        d.emit('httpRequest' => message)
+        d.run
+      end
+      verify_log_entries(1, COMPUTE_PARAMS, 'httpRequest') do |entry|
+        assert_equal http_request_message_expected_without_referer,
+                     entry['httpRequest'], entry
+        assert_equal expected,
+                     get_fields(entry['structPayload'])['httpRequest'].to_json,
```

Comment:
Sorry, are we sure this is the right behavior? Should we really be leaving a struct with a `null` referer in this case? Or should we be deleting the referer and ending up with a `nil` struct?